source/common/network/io_socket_handle_impl.cc

```diff
@@ -48,6 +48,24 @@ Api::IoCallUint64Result IoSocketHandleImpl::readv(uint64_t max_length, Buffer::R
       fd_, iov.begin(), static_cast<int>(num_slices_to_read)));
 }
 
+Api::IoCallUint64Result IoSocketHandleImpl::read(Buffer::Instance& buffer, uint64_t max_length) {
```

Comment:
Would be good to have more comprehensive IoSocketHandleImpl tests in test/common/network/io_socket_handle_impl_test.cc

Right now we rely on test/common/buffer/owned_impl_test.cc to provide test coverage.

Reply:
I guess we could move some of the tests if we manage to change the write side as well?

Reply:
My observation is that there are no tests for some of the other operations in IoSocketHandleImpl like readv or sendmsg.   It is true that some of the methods are trivial, but both readv and sendmsg look fairly long and complex.  readv for instance does some adjustments to the provided RawSlices as it converts them to iovec and applies a max_length restriction.

Reply:
Ow, I went in a different direction! Yeah, we should eventually add more tests to io_socket_handle_impl_test.cc

Reply:
The test coverage is not really changing much with this PR, so I'm fine with not blocking on additional tests. It would be good to prioritize general IoSocketHandleImpl test coverage in some future PR.